R/functions-range.R

```diff
@@ -219,3 +220,171 @@ pl__datetime_ranges <- function(
     )
   })
 }
+
+#' Generate a range of integers
+#'
+#' @inheritParams rlang::check_dots_empty0
+#' @param start Start of the range (inclusive). Defaults to 0.
+#' @param end End of the range (exclusive). If `NULL` (default), the value
+#' of `start` is used and `start` is set to 0.
+#' @param step Step size of the range.
+#' @param dtype Data type of the range.
+#'
+#' @inherit as_polars_expr return
+#' @examples
+#' pl$select(int = pl$int_range(0, 3))
+#'
+#' # end can be omitted for a shorter syntax.
+#' pl$select(int = pl$int_range(3))
+#'
+#' # Generate an index column by using int_range in conjunction with len().
+#' df <- pl$DataFrame(a = c(1, 3, 5), b = c(2, 4, 6))
+#' df$select(
+#'   index = pl$int_range(pl$len(), dtype = pl$UInt32),
+#'   pl$all()
+#' )
+pl__int_range <- function(
+    start = 0,
+    end = NULL,
+    step = 1,
+    ...,
+    dtype = pl$Int64) {
+  wrap({
+    check_dots_empty0(...)
+    if (is.null(end)) {
+      end <- start
+      start <- 0
+    }
+    int_range(
+      as_polars_expr(start)$`_rexpr`,
+      as_polars_expr(end)$`_rexpr`,
+      step,
+      dtype$`_dt`
```

Comment:
`dtype`'s type should be checked first. (`` dtype$`_dt` `` should be error)

Reply:
Done